### PR TITLE
Fixed failed tools tests when source directory is read-only.

### PR DESCRIPTION
**Pull request by nmoinvaz**
See #1232.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1209,7 +1209,7 @@ if(ZLIB_ENABLE_TESTS)
     add_test(NAME makefixed
         COMMAND ${CMAKE_COMMAND}
         "-DCOMMAND=${MAKEFIXED_COMMAND}"
-        -DOUTPUT=${CMAKE_CURRENT_BINARY_DIR}/inffixed_tbl._h
+        -DOUTPUT=${CMAKE_CURRENT_BINARY_DIR}/Testing/Temporary/inffixed_tbl._h
```
Comment (mtl1979) [inline CMakeLists.txt:1212]: This doesn't make sense at all... `CMAKE_CURRENT_BINARY_DIR` is never supposed to be read-only.

Comment (nmoinvaz) [reply CMakeLists.txt:1212]: Good point, perhaps if `CMAKE_CURRENT_BINARY_DIR==CMAKE_CURRENT_SOURCE_DIR`?

**Comment by mtl1979** [comment]
See https://github.com/zlib-ng/zlib-ng/blob/develop/cmake/run-and-compare.cmake#L44

When `COMPARE` points to `CMAKE_CURRENT_SOURCE_DIR` and we're doing out-of-tree build, it needs to be replaced with `CMAKE_CURRENT_BINARY_DIR`.

File: CMakeLists.txt (hunk 1 above)
Comment (mtl1979) [reply CMakeLists.txt:1212]: Obviously we need to check if we are doing out-of-tree build... See my previous comment...

**Comment by nmoinvaz** [comment]
@mtl1979 should be fixed now. Good catch!

**Comment by mtl1979** [comment]
> @mtl1979 should be fixed now. Good catch!

Usually I can see what's going wrong, but I'm not really big fan of learning all the cmake functions because they keep changing them in every version...

**Comment by phprus** [comment]
This PR fixes issue #1232. Thanks!

**Review by Dead2 — APPROVED** [review]

**Pull request merged**